cli/tests/integration/lsp_tests.rs

```diff
@@ -3170,6 +3170,77 @@ fn lsp_format_markdown() {
   shutdown(&mut client);
 }
 
+#[test]
+fn lsp_format_with_config() {
+  let temp_dir = TempDir::new().expect("could not create temp dir");
+  let mut params: lsp::InitializeParams =
+    serde_json::from_value(load_fixture("initialize_params.json")).unwrap();
+  let deno_fmt_jsonc =
+    serde_json::to_vec_pretty(&load_fixture("deno.fmt.jsonc")).unwrap();
+  fs::write(temp_dir.path().join("deno.fmt.jsonc"), deno_fmt_jsonc).unwrap();
+
+  params.root_uri = Some(Url::from_file_path(temp_dir.path()).unwrap());
+  if let Some(Value::Object(mut map)) = params.initialization_options {
+    map.insert("config".to_string(), json!("./deno.fmt.jsonc"));
+    params.initialization_options = Some(Value::Object(map));
+  }
+
+  let deno_exe = deno_exe_path();
+  let mut client = LspClient::new(&deno_exe).unwrap();
+  client
+    .write_request::<_, _, Value>("initialize", params)
+    .unwrap();
+
+  client
+    .write_notification(
+      "textDocument/didOpen",
+      json!({
+        "textDocument": {
+          "uri": "file:///a/file.ts",
+          "languageId": "typescript",
+          "version": 1,
+          "text": "export async function someVeryLongFunctionName() {\nconst response = fetch(\"http://localhost:4545/some/non/existent/path.json\");\nconsole.log(response.text());\nconsole.log(\"finished!\")\n}"
+        }
+      }),
+    )
+    .unwrap();
+
+  let (maybe_res, maybe_err) = client
+    .write_request::<_, _, Value>(
+      "textDocument/formatting",
+      json!({
+        "textDocument": {
+          "uri": "file:///a/file.ts"
+        },
+        "options": {}
+      }),
```

Comment:
I have trouble with this test, I'm getting following error:
```
running 1 test
test integration::lsp::lsp_format_with_config ... FAILED

failures:

---- integration::lsp::lsp_format_with_config stdout ----
Some(
    LspResponseError {
        code: -32602,
        message: "missing field `tabSize`",
        data: None,
    },
)
thread 'integration::lsp::lsp_format_with_config' panicked at 'assertion failed: maybe_err.is_none()', cli/tests/integration/lsp_tests.rs:3221:3
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

```

Should I just pass fake data here and count on the fact that config file should take precedence over these options?

CC @kitsonk

Reply:
I think that's fine to provide some fake data. The editor will have a "tab size" for a file that might not be synced with the formatting settings.